lib/encoder.js

```diff
@@ -0,0 +1,68 @@
+const abi = require('web3-eth-abi')
+const namehash = require('eth-ens-namehash')
+const ACL_ABI = require('../artifacts/Acl.json').abi
+const AGENT_ABI = require('../artifacts/Agent.json').abi
+const ENS_ABI = require('../artifacts/ENS.json').abi
+const FINANCE_ABI = require('../artifacts/Finance.json').abi
+const FORWARDER_ABI = require('../artifacts/IForwarderWithContext.json').abi
+const KERNEL_ABI = require('../artifacts/Kernel.json').abi
+const REWARDS_ABI = require('../artifacts/Rewards.json').abi
+
+const CALLSCRIPT_ID = '0x00000001'
+const EMPTY_HEX = '0x00'
+
+function encodeCallsScript(actions) {
+  return actions.reduce((script, { to, data }) => {
+    const address = abi.encodeParameter('address', to)
+    const dataLength = abi.encodeParameter('uint256', (data.length - 2) / 2).toString('hex')
+    return script + address.slice(26) + dataLength.slice(58) + data.slice(2)
+  }, CALLSCRIPT_ID)
+}
+
+function encodeInvalidTransaction(ensContract, ensDomain, targetAgent) {
```

Comment:
What about renaming this one to `encodeSetEnsOwnerThroughAgent` or similar?

Reply:
See https://github.com/aragon/court-precedence-campaign/pull/3#discussion_r480154593